flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -409,6 +410,12 @@ public TaskExecutor(
                 resourceId, new JobManagerHeartbeatListener(), getMainThreadExecutor(), log);
     }
 
+    private boolean isJobRecoveryEnabled() {
```

Comment:
Better to rename it to `shoudRetainPartitionsOnJobManagerConnectionLost`.
And it should return `true` only if netty shuffle is used.